tests/functional/flakes/show.sh

```diff
@@ -6,10 +6,25 @@ mkdir -p "$flakeDir"
 writeSimpleFlake "$flakeDir"
 cd "$flakeDir"
 
+# Check the flake reference that was produced.
+nix flake show --json > output.json
+nix eval --impure --expr '
+with builtins;
+let
+  flakeRef = parseFlakeRef (head (attrNames (fromJSON (readFile ./output.json))));
+  narHash = if match ".*darwin" currentSystem != null
+    then "sha256-MhmXWT9U41eDFezA02mAj3PxBO21icT5VvvPSC1epDU="
```

Comment:
This test fails on aarch64-darwin, and will fail on any platform except x86-64_darwin and x86-64_linux.
I would recommend changing it to something that only checks the format of the flakeRef, not the hash itself.